test/models/post_test.rb

```diff
@@ -1,7 +1,23 @@
 require 'test_helper'
 
 class PostTest < ActiveSupport::TestCase
-  # test "the truth" do
-  #   assert true
-  # end
+  test "defaults opengraph fields to nil when no links in body text" do
+    post = FactoryGirl.create(:post)
+    assert_nil post.og_url
+    assert_nil post.og_title
+    assert_nil post.og_description
+    assert_nil post.og_image
+  end
+
+  test "grabs opengraph data from site url" do
+    # will need to update this in the future if the og metadata tags on the site change
+    # currently a helper method tacks on http to the url, maybe test that separately?
+    post = FactoryGirl.create(:post, body: "How about www.thefirehoseproject.com?")
+    assert_equal "http://www.thefirehoseproject.com", post.og_url
+    assert_equal "Mentor Driven Online Coding Bootcamp", post.og_title
+    assert_equal "Get 1-on-1 mentor sessions, join an agile team project "\
+    "and be ready to switch careers into web development or launch your "\
+    "own idea.", post.og_description
+    assert_equal "http://www.thefirehoseproject.com/assets/meta-tag-coder.png", post.og_image
```

Comment:
The code here is good, except that it looks like our test is actually making an HTTP request to the web.  That means if I change the meta tags on our site the test will fail, even though the code is the same.  This is a bad thing.  Take a look at [FakeWeb](https://github.com/chrisk/fakeweb).  I like to use that gem to make sure errors happen on real HTTP requests:

```
FakeWeb.allow_net_connect = false
```